src/Sylius/Bundle/ShopBundle/Controller/ContactController.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ShopBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\EmailManager\ContactEmailManager;
+use Sylius\Bundle\CoreBundle\Form\Type\ContactType;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ContactController extends Controller
+{
+    /**
+     * @var ContactEmailManager
+     */
+    private $contactEmailManager;
+
+    /**
+     * @param ContactEmailManager $contactEmailManager
+     */
+    public function __construct(ContactEmailManager $contactEmailManager)
+    {
+        $this->contactEmailManager = $contactEmailManager;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function requestAction(Request $request)
+    {
+        $formType = $this->getSyliusAttribute($request, 'form', ContactType::class);
+        $form = $this->createForm($formType);
+
+        if ($request->isMethod('POST') && $form->handleRequest($request)->isValid()) {
+            $data = $form->getData();
+
+            if (!$this->contactEmailManager->sendContactRequest($data)) {
```

Comment:
Right now we're injecting `ChannelContext` to `ContactEmailManager`. I think the better choice is to get the channel from the context in the controller and then pass it as an argument in `sendContactRequest`.